src/llvm-late-gc-lowering.cpp

```diff
@@ -171,39 +173,54 @@ static std::pair<Value*,int> FindBaseValue(const State &S, Value *V, bool UseCac
             (void)LI;
             break;
         }
-        else if (auto II = dyn_cast<IntrinsicInst>(CurrentV)) {
+        else if (dyn_cast<IntrinsicInst>(CurrentV) != nullptr &&
+                (cast<IntrinsicInst>(CurrentV)->getIntrinsicID() == Intrinsic::masked_load ||
+                cast<IntrinsicInst>(CurrentV)->getIntrinsicID() == Intrinsic::masked_gather)) {
```

Comment:
This code pattern is annoying since you are doing three casts.  

Can we restore the previous code pattern?

Reply:
The old pattern was wrong, it was catching every intrinsic and infinite looping, we should only enter the branch if we are going to handle it. Maybe this is a use for a goto